crates/router/src/core/payments/helpers.rs

```diff
@@ -939,7 +999,8 @@ pub fn create_complete_authorize_url(
 }
 
 fn validate_recurring_mandate(req: api::MandateValidationFields) -> RouterResult<()> {
-    req.mandate_id.check_value_present("mandate_id")?;
+    req.recurring_details
+        .check_value_present("recurring_details")?;
```

Comment:
we need to check `mandate_id` also right? One of the two should be given.

Reply:
`recurring_details` is of type `RecurringDetails` in which during from conversion itself, I'm mapping the mandate_id to `RecurringDetails::MandateId` or else take the request.recurring_details

```
impl From<&PaymentsRequest> for MandateValidationFields {
    fn from(req: &PaymentsRequest) -> Self {
        let recurring_details = req
            .mandate_id
            .clone()
            .map(RecurringDetails::MandateId)
            .or(req.recurring_details.clone());

        Self {
            recurring_details,
            confirm: req.confirm,
            customer_id: req
                .customer
                .as_ref()
                .map(|customer_details| &customer_details.id)
                .or(req.customer_id.as_ref())
                .map(ToOwned::to_owned),
            mandate_data: req.mandate_data.clone(),
            setup_future_usage: req.setup_future_usage,
            off_session: req.off_session,
        }
    }
}
```